app/src/main/java/cm/aptoide/pt/account/view/ImageValidator.kt

```diff
@@ -0,0 +1,67 @@
+package cm.aptoide.pt.account.view
+
+import cm.aptoide.pt.account.view.exception.InvalidImageException
+import cm.aptoide.pt.account.view.exception.InvalidImageException.ImageError
+import rx.Completable
+import rx.Scheduler
+import java.util.*
+
+open class ImageValidator(private val scheduler: Scheduler,
+                          private val imageInfoProvider: ImageInfoProvider) {
+
+  open fun validateOrGetException(imagePath: String): Completable? {
+    return Completable.defer {
+      validate(imagePath)
+      return@defer Completable.complete()
+    }
+        .subscribeOn(scheduler)
+  }
+
+  @Throws(InvalidImageException::class)
+  private fun getInfo(imagePath: String): ImageInfo {
+    val info = imageInfoProvider.getInfo(imagePath)
+    if (info == null) {
+      val errors: MutableList<ImageError> = ArrayList()
+      errors.add(ImageError.ERROR_DECODING)
+      throw InvalidImageException(errors)
+    }
+    return info
+  }
+
+  @Throws(InvalidImageException::class)
+  private fun validate(imagePath: String) {
+    val imageInfo = getInfo(imagePath)
+    val errors: MutableList<ImageError> = ArrayList()
+    if (imageInfo == null) {
+      errors.add(ImageError.ERROR_DECODING)
+    } else {
+      if (imageInfo.height < MIN_IMAGE_HEIGHT) {
+        errors.add(ImageError.MIN_HEIGHT)
```

Comment:
This can be if else if, instead of 4 ifs.
if(imageInfo.height < MIN_IMAGE_HEIGHT){
...
}else if(imageInfo.height > MAX_IMAGE_HEIGHT){
...
}